File: app/src/main/java/com/walhalla/appextractor/activity/resources/p002/ManifestContract2.java
package com.walhalla.appextractor.activity.resources.p002;

import com.walhalla.appextractor.activity.resources.ResItem;

import java.util.List;

public interface ManifestContract2 {
    interface View {
        void showManifestContent(String content);
        void showError(String message);

        void showSuccess(List<ResItem> list);
    }

    interface Presenter {
        void loadManifestContent(String packageName);
    }
}

File: app/src/main/java/com/walhalla/appextractor/activity/resources/p002/ManifestContract2.kt
package com.walhalla.appextractor.activity.resources.p002

import com.walhalla.appextractor.resources.ResItem

interface ManifestContract2 {
    interface View {
        fun showManifestContent(content: String)
        fun showError(message: String)

        fun showSuccess(list: List<ResItem>)
    }

    open interface Presenter {
        fun loadManifestContent(packageName: String?)
    }
}
